authentication/src/main/java/io/camunda/authentication/handler/session/SessionDocumentMapper.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class SessionDocumentMapper {
+
+  public static final String POLLING_HEADER = "x-is-polling";
+  private static final Logger LOGGER = LoggerFactory.getLogger(SessionDocumentMapper.class);
+  private final GenericConversionService conversionService;
+
+  public SessionDocumentMapper(final GenericConversionService conversionService) {
+    this.conversionService = conversionService;
+  }
+
+  @PostConstruct
+  private void setUp() {
+    setupConverter();
+  }
+
+  public void setupConverter() {
+    conversionService.addConverter(Object.class, byte[].class, new SerializingConverter());
+    conversionService.addConverter(byte[].class, Object.class, new DeserializingConverter());
+  }
+
+  public Map<String, Object> sessionToDocument(final WebSession session) {
+    final Map<String, byte[]> attributes = new HashMap<>();
+    session
+        .getAttributeNames()
+        .forEach(name -> attributes.put(name, serialize(session.getAttribute(name))));
+    return Map.of(
+        ID,
+        session.getId(),
+        CREATION_TIME,
+        session.getCreationTime().toEpochMilli(),
+        LAST_ACCESSED_TIME,
+        session.getLastAccessedTime().toEpochMilli(),
+        MAX_INACTIVE_INTERVAL_IN_SECONDS,
+        session.getMaxInactiveInterval() == null
+            ? Long.MAX_VALUE
+            : session.getMaxInactiveInterval().getSeconds(),
+        ATTRIBUTES,
+        attributes);
+  }
+
+  public WebSession documentToSession(
+      final HttpServletRequest request, final Map<String, Object> document) {
+    try {
+      if (document == null || document.isEmpty()) {
+        LOGGER.warn("Document is empty");
+        return null;
+      }
+
+      final String sessionId = getSessionIdFrom(document);
+      final WebSession session = new WebSession(sessionId);
+      session.setLastAccessedTime(getInstantFor(document.get(LAST_ACCESSED_TIME)));
+      try {
+        if (request != null && request.getHeader(POLLING_HEADER) != null) {
+          request.getHeader(POLLING_HEADER);
+          session.setPolling(true);
+        }
+      } catch (final Exception ignored) {
+        // It's not a polling request
+      }
+      session.setCreationTime(getInstantFor(document.get(CREATION_TIME)));
+      session.setMaxInactiveInterval(
+          getDurationFor(document.get(MAX_INACTIVE_INTERVAL_IN_SECONDS)));
+
+      final Object attributesObject = document.get(ATTRIBUTES);
+      if (attributesObject instanceof Map) {
+        final Map<String, Object> attributes = (Map<String, Object>) document.get(ATTRIBUTES);
+        attributes
+            .keySet()
+            .forEach(
+                name -> {
+                  if (attributes.get(name) instanceof String) {
+                    session.setAttribute(
+                        name,
+                        deserialize(Base64.getDecoder().decode((String) attributes.get(name))));
```

Comment:
🔧  Should we add a try/catch in order to catch possible deserialisation errors over here and log that?